configs/components/puppet.rb

```diff
@@ -251,28 +251,44 @@
     env_hiera = File.join(settings[:puppet_codedir], 'environments', 'production', 'hiera.yaml')
     rmv_hiera = File.join(configdir, 'remove_hiera5_files.rm')
 
-    # Pre-Install script to save copies of existing hiera configuration files.
-    # This also creates "marker files to indicate that the files pre-existed."
+    # Pre-Install script saves copies of existing hiera configuration files,
+    # and also creates "marker" files to indicate that the files pre-existed.
+    # We want to ensure that our copy preserves the metadata (e.g. permissions,
+    # timestamps, extended attributes) of the existing hiera config. files. The
+    # necessary flags that do this are platform dependent.
+    if platform.is_solaris?
+      # The @ option preserves extended attributes
+      cp_cmd = "cp -p@"
```

Comment:
So unfortunately, it seems like `cp` will always update the `changed` timestamp -- I could not find a flag that would avoid this in any of the manpages. If we really want to preserve everything, I think we could try something like:

(1) Rename the hiera files to adopt the pkg-old suffix
(2) Copy the pkg-old suffix files to their original paths

^ This way, we preserve the permissions and all of the timestamps when moving pkg-old back in the post-installation script. Only problem is if the upgrade fails, then the user is left with hiera.yaml files that have their "changed" timestamps updated.

Reply:
I think I'm fine blowing away the timestamps